include/aws/common/logging.h

```diff
@@ -0,0 +1,117 @@
+#ifndef AWS_COMMON_LOGGING_H
+#define AWS_COMMON_LOGGING_H
+
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/common.h>
+
+#define AWS_LOG_LEVEL_NONE 0
```

Comment:
What's the reason for having these redundantly declared as macros and an enum? The enum could just have these values in it and cover everything?

Reply:
In order to do the preprocessor-level filter comparison we must have the value #defined.  The enum isn't strictly necessary, but I kind of prefer it as the user-facing type.  I don't think it adds any type safety though in C?